newton/_src/viewer/viewer_file.py

```
@@ -72,24 +1090,126 @@ def log_state(self, state):
+        self._save_recording(effective_path, verbose=verbose)
+
+    def _save_recording(self, file_path: str | None = None, verbose: bool = True):
         """Internal method to save recording."""
         try:
-            self._recorder.save_to_file(str(self.output_path))
-            print(f"Recording saved to {self.output_path} ({self._frame_count} frames)")
+            effective_path = file_path if file_path is not None else str(self.output_path)
+            self._save_to_file(effective_path)
+            if verbose:
+                print(f"Recording saved to {effective_path} ({self._frame_count} frames)")
         except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)